### PR TITLE
Lock rotation of blocks with `rotate = false` to 0

### PR DESCRIPTION
**Pull request by MEEPofFaith**
https://user-images.githubusercontent.com/54301439/233856676-12391a82-cae1-456d-9329-0dd6bb485963.mp4

It's on a boolean because apparently some mods use the fact that blocks that don't rotate still rotate.

If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [X] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [X] I have ensured that my code compiles, if applicable.
- [X] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

File: core/src/mindustry/world/Block.java [hunk 1]
```diff
@@ -106,6 +106,8 @@ public class Block extends UnlockableContent implements Senseable{
     public boolean underBullets;
     /** whether this is rotatable */
     public boolean rotate;
+    /** if rotate is false and this is true, building rotation is set to 0 regardless of player input */
+    public boolean lockRotation = true;
```
Comment (MEEPofFaith) [inline core/src/mindustry/world/Block.java:110]: Should it default to false?

Comment (BlueWolf3682) [reply core/src/mindustry/world/Block.java:110]: probably not, PRed values probably shouldnt force changes that may affect vanilla if intentionally avoidable

**Comment by sk7725** [comment]
I don't see any use with this field. If you want to lock rotation that bad, you can just set the rotation to 0 on block initialization for individual blocks (via Java/JS).
Json mods have no use whatsoever of this block, either.

**Comment by sk7725** [comment]
> It's on a boolean because apparently some mods use the fact that blocks that don't rotate still rotate.

afaik vanilla hortiloaders use this too.
To specify, mods are not using the fact that it "still rotates" - mods are using the fact the initial rotation tells you how (and in what direction) the block was placed, even when rotate = false.

**Comment by MEEPofFaith** [comment]
> I don't see any use with this field. If you want to lock rotation that bad, you can just set the rotation to 0 on block initialization for individual blocks (via Java/JS). Json mods have no use whatsoever of this block, either.

Case 1: Directional liquid outputs on a generic crafter that has `rotate = false`. Json has no way to disable.
Case 2: Setting the rotation on placement will not visually show it in building highlights. `IntputHandler` drawing block previews will still rotate the block, leading to improper displays for rotation-based things such as generic crafter liquid outputs.

**Comment by sk7725** [comment]
> > I don't see any use with this field. If you want to lock rotation that bad, you can just set the rotation to 0 on block initialization for individual blocks (via Java/JS). Json mods have no use whatsoever of this block, either.
> 
> Case 1: Directional liquid outputs on a generic crafter that has `rotate = false`. Json has no way to set rotation to 0 on placement. Case 2: Setting the rotation on placement will not visually show it in building highlights. `IntputHandler` drawing block previews will draw a rotated preview, leading to improper displays for rotation-based things such as generic crafter liquid outputs.

Case 1: set `rotate = true` for json.
Case 2: override the preview draw functions.

**Comment by BlueWolf3682** [comment]
this is a lot of complaining about useless additions from the person that added placeablePlayer

**Comment by Anuken** [comment]
The setRotation change seems too invasive, I don't want to use a getter everywhere.

**Comment by MEEPofFaith** [comment]
god damnit

**Comment by MEEPofFaith** [comment]
why are the power gen tests failing what

**Comment by Anuken** [comment]
I broke it

**Comment by sk7725** [comment]
planRotation() sounds better, I am still skeptical that this will have an use out of very niche scenarios

**Comment by sk7725** [comment]
> this is a lot of complaining about useless additions from the person that added placeablePlayer

I only complain about new field additions that can be achievable by using existing code workarounds. I'm cool with a field that can introduce new behaviors.

**Comment by MEEPofFaith** [comment]
> 

ive never even heard of `placablePlayer` does anyone even use it

**Comment by MEEPofFaith** [comment]
> > this is a lot of complaining about useless additions from the person that added placeablePlayer
> 
> I only complain about new field additions that can be achievable by using existing code workarounds. I'm cool with a field that can introduce new behaviors.

ive never even heard of `placablePlayer` does anyone even use it

File: core/src/mindustry/world/blocks/production/GenericCrafter.java [hunk 2]
```diff
@@ -118,6 +118,8 @@ public void init(){
 
         if(outputItems != null) hasItems = true;
         if(outputLiquids != null) hasLiquids = true;
+        
+        if(!rotate) lockRotation = true;
```
Comment (MEEPofFaith) [inline core/src/mindustry/world/blocks/production/GenericCrafter.java:122]: Should this be included?

Comment (Anuken) [reply core/src/mindustry/world/blocks/production/GenericCrafter.java:122]: No, that looks redundant.

**Comment by MEEPofFaith** [comment]
See convo in #pulls.
If a block has `rotate = false`, it doesn't make sense for it to rotate anyway. That is a bug and unintended behavior. If you want to not show rotation, then just do `drawArrow = false` instead.

**Pull request merged**